experimental/unitsconversion/src/converter.rs

```diff
@@ -0,0 +1,161 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::{
+    measureunit::{MeasureUnit, MeasureUnitParser},
+    provider::{MeasureUnitItem, UnitsInfoV1},
+    ConversionError,
+};
+use litemap::LiteMap;
+use zerotrie::ZeroTrieSimpleAscii;
+use zerovec::{ZeroSlice, ZeroVec};
+
+/// Represents the possible cases for the convertibility between two units.
+pub enum Convertibility {
+    /// The units are convertible.
+    /// For example, `meter` and `foot` are convertible.
+    Convertible,
+
+    /// The units are reciprocal.
+    /// For example, `gallon-per-mile` and `100-kilometer-per-liter` are reciprocal.
+    Reciprocal,
+
+    /// The units are not convertible.
+    /// For example, `meter` and `second` are not convertible.
+    NotConvertible,
+}
+
+/// A factory for creating a converter.
+/// Also, it can check the convertibility between two units.
+pub struct ConverterFactory<'data> {
+    /// Contains the necessary data for the conversion factory.
+    payload: &'data UnitsInfoV1<'data>,
```

Comment:
this should use a `DataPayload` and own the data, yes?

Reply:
copy PluralRules or ListFormatter

Reply:
I added a TODO. #4522